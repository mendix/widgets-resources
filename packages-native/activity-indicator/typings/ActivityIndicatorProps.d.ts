/**
 * This file was generated from ActivityIndicator.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix Widgets Team
 */

interface CommonProps<Style> {
    name: string;
    style: Style[];
}

export interface ActivityIndicatorProps<Style> extends CommonProps<Style> {}
