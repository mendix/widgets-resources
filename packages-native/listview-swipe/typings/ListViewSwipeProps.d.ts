/**
 * This file was generated from ListViewSwipe.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix Widgets Team
 */
import { ActionValue } from "mendix";

interface CommonProps<Style> {
    style: Style[];
}

export type LeftRenderModeEnum = "disabled" | "action" | "buttons";

export type RightRenderModeEnum = "disabled" | "action" | "buttons";

export interface ListViewSwipeProps<Style> extends CommonProps<Style> {
    content: any;
    leftRenderMode: LeftRenderModeEnum;
    left?: any;
    onSwipeLeft?: ActionValue;
    closeOnFinishLeft: boolean;
    rightRenderMode: RightRenderModeEnum;
    right?: any;
    onSwipeRight?: ActionValue;
    closeOnFinishRight: boolean;
}
