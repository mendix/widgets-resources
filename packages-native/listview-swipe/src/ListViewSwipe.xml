<?xml version="1.0" encoding="utf-8"?>
<widget id="com.mendix.widget.native.listviewswipe.ListViewSwipe" supportedPlatform="Native" needsEntityContext="true"
        offlineCapable="true" pluginWidget="true" xmlns="http://www.mendix.com/widget/1.0/"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../../../node_modules/mendix/custom_widget.xsd">
    <name>List View Swipe</name>
    <description/>
    <icon>iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAYAAACqaXHeAAAABGdBTUEAALGPC/xhBQAAADhlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAAqACAAQAAAABAAAAQKADAAQAAAABAAAAQAAAAABlmWCKAAADxUlEQVR4Ae1aP2siQRR/SgqFHFoYEkKusZDUfgUrQxCbxOIkZbDyE+QziGCqlAdXKGJxRXJFqpAidUhSpRQRFEXO4hDU29m4y+4+d9+4u+PuJjsQMjPvzfvzmzf/ngsQlhCBEIGvjEDE6HypVMovFosbqf/ISAtYuxuNRi+bzeadld1RI/GTOM/cOlr5YnRR10YA6KjBb5BRjABgYSP53Q2+73weoD2Ab5j7XGdnZ0ut1L8/fmqbtut/inFLH1EE2NYU0IE7lN353//ysFjeSNOzdj1J8HYhGrm8K8Qsd1tKj1d0OgIsnGdGy8BIPF454FQvCYDZzGsV8/Bo+f1UJwHwk7EibAkBEIFqkGSSESDv8oRHPDyECM/IJADsiLNyUDkGPfPAoWLyHrA637871OPb4XQE+NZ0dwxDEeCXfMC3XxfueEhIQRHwifIBhOsfZAQA16jgMJHPegTAJ8oHyCmx4MxVaKk3CEj3GOvidj7Aw1NmbZYY7QEIDpfzAR6eMmuzxCQAPG99Hh4NsGszSxq6yCrSTQIg0ho/yEY3wW0b1Wq1hKo8Pz+3lP/lI4AEwOoprEDLw6Pw+u0/vQRYPsDiJHA7H1Cr1VSM0uk0FItFuf3+/g739/fw+voK4/EYdnd34fDwEDKZjMwTi8XUcZtUSAC2nQ94enpS7X95eYFCoQDtdhs6nQ5IR6hKG41GwP4Yz8PDA1SrVTg+PlbpvBVyCfAKEsE3nU7h+vpaBkDrvFHXcDiERqMBs9nMSCLbKAIE3NTW3sBIy1YMj4+PKmskEoFEIgGTyQSWS+n2oSmDwQBub2/VJaMhWVYRAAJuasoNzHZaTXqhQrlchlwuB/F4HFhk1Ot1eH5+1jn39va2MQC+XgKKd8z509NT2XnWxzbASqUCDBht6ff72iZXXS9BGiIgH+DoXb63twcnJyfImVQqBQcHB7p+tgw2LWgJrL6psR2umxpA8bOwN860MiaZTEKv11OaMJ/P1TpvBUUA78Bt8WWzWVNVOzto/kx5zQikBLfzAWaGmPXv7++bkVzppyPA4hbILJCfwgK/D2C7vshCAsDz1ufhEemEE9kkAE6EB2FsCEAQZkmkjWQE8Lz1eXhEOuFENglA+H3Ax/d/nt0M2auPvQJFFfIiJEqxmdxNkqRXV1dmYrj7EQB+ywdwe2KTEe0BAvMBNk0UOwxFgFh1WDqVt8cj3O1BEeC3fIC77mJpKAK2kA/oSmag3+iwaUJ6mG5dQRGgowpoCIgwXisdZaZ4lYR8IQIhAiECIQJBQuA/4awjBiNddXEAAAAASUVORK5CYII=</icon>
    <properties>
        <propertyGroup caption="General">
            <property key="content" type="widgets">
                <caption>Foreground</caption>
                <description>Sample text input</description>
            </property>
        </propertyGroup>
        <propertyGroup caption="Left to Right">
            <propertyGroup caption="Swipe">
                <property key="left" type="widgets" required="false">
                    <caption>Left background</caption>
                    <description/>
                </property>
                <property key="leftRenderMode" type="enumeration" defaultValue="disabled">
                    <caption>Pattern</caption>
                    <description>"Show options" will show the background buttons. "Toggle" animates the item back to the initial state. "Swipe out and reset" animates the item out of the list.</description>
                    <enumerationValues>
                        <enumerationValue key="disabled">Disabled</enumerationValue>
                        <enumerationValue key="buttons">Show options (buttons)</enumerationValue>
                        <enumerationValue key="archive">Swipe out and collapse</enumerationValue>
                        <enumerationValue key="swipeOutReset">Swipe out and reset</enumerationValue>
                        <enumerationValue key="toggle">Toggle</enumerationValue>
                    </enumerationValues>
                </property>
            </propertyGroup>
            <propertyGroup caption="Events">
                <property key="onSwipeLeft" type="action" required="false">
                    <caption>On swipe action</caption>
                    <description>Action triggered for pattern: "Swipe out and collapse" after the item is collapsed, "Toggle" when item passed the threshold, "Swipe out and reset" after the item is animated out of the list.</description>
                </property>
            </propertyGroup>
        </propertyGroup>
        <propertyGroup caption="Right to Left">
         <propertyGroup caption="Swipe">
                <property key="right" type="widgets" required="false">
                    <caption>Right background</caption>
                    <description/>
                </property>
                <property key="rightRenderMode" type="enumeration" defaultValue="disabled">
                    <caption>Pattern</caption>
                    <description>"Show options" will show the background buttons. "Toggle" animates the item back to the initial state. "Swipe out and reset" animates the item out of the list.</description>
                    <enumerationValues>
                        <enumerationValue key="disabled">Disabled</enumerationValue>
                        <enumerationValue key="buttons">Show options (buttons)</enumerationValue>
                        <enumerationValue key="archive">Swipe out and collapse</enumerationValue>
                        <enumerationValue key="swipeOutReset">Swipe out and reset</enumerationValue>
                        <enumerationValue key="toggle">Toggle</enumerationValue>
                    </enumerationValues>
                </property>
            </propertyGroup>
            <propertyGroup caption="Events">
                <property key="onSwipeRight" type="action" required="false">
                    <caption>On swipe action</caption>
                    <description>Action triggered for pattern: "Swipe out and collapse" after the item is collapsed, "Toggle" when item passed the threshold, "Swipe out and reset" after the item is animated out of the list.</description>
                </property>
            </propertyGroup>
        </propertyGroup>
    </properties>
</widget>
