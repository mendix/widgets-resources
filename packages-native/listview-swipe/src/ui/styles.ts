import { ViewStyle } from "react-native";
import { Style } from "@native-mobile-resources/util-widgets";

export interface PanelStyle extends ViewStyle {
    panelSize: number;
}

export interface ListViewSwipeStyle extends Style {
    container: ViewStyle;
    leftAction: PanelStyle;
    rightAction: PanelStyle;
}

export const defaultListViewSwipeStyle: ListViewSwipeStyle = {
    container: {
        flex: 1,
        height: 80,
        paddingVertical: 10,
        paddingHorizontal: 20,
        justifyContent: "space-between",
        flexDirection: "column",
        backgroundColor: "white"
    },
    leftAction: {
        flexDirection: "row",
        alignItems: "stretch",
        flex: 1,
        justifyContent: "center",
        panelSize: 192
    },
    rightAction: {
        flexDirection: "row",
        alignItems: "stretch",
        flex: 1,
        justifyContent: "center",
        panelSize: 192
    }
};
