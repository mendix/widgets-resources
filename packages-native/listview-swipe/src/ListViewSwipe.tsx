import { Component, createElement, ReactNode } from "react";
import { ListViewSwipeProps } from "../typings/ListViewSwipeProps";
import { flattenStyles } from "@native-mobile-resources/util-widgets";
import { View } from "react-native";
import { RectButton } from "react-native-gesture-handler";
import Swipeable from "react-native-gesture-handler/Swipeable";
import { defaultListViewSwipeStyle, ListViewSwipeStyle, PanelStyle } from "./ui/styles";

export class ListViewSwipe extends Component<ListViewSwipeProps<ListViewSwipeStyle>> {
    private row: any;
    private readonly styles = flattenStyles(defaultListViewSwipeStyle, this.props.style);
    private readonly onSwipeLeftHandler = this.onSwipeLeft.bind(this);
    private readonly onSwipeRightHandler = this.onSwipeRight.bind(this);

    private readonly isLeftSideAction = this.props.leftRenderMode === "action";
    private readonly isRightSideAction = this.props.rightRenderMode === "action";
    private readonly isLeftDisabled = this.props.leftRenderMode === "disabled";
    private readonly isRightDisabled = this.props.rightRenderMode === "disabled";

    render(): ReactNode {
        return (
            <Swipeable
                ref={this.updateRef}
                friction={1}
                leftThreshold={this.isLeftSideAction ? 100 : 40}
                rightThreshold={this.isRightSideAction ? 100 : 40}
                renderLeftActions={this.renderLeftActions}
                renderRightActions={this.renderRightActions}
                onSwipeableLeftOpen={this.onSwipeLeftHandler}
                onSwipeableRightOpen={this.onSwipeRightHandler}
                overshootLeft={false}
                overshootRight={false}
                useNativeAnimations
            >
                <View style={this.styles.container}>{this.props.content}</View>
            </Swipeable>
        );
    }

    renderLeftActions = (): ReactNode => {
        if (this.isLeftDisabled) {
            return undefined;
        } else if (this.isLeftSideAction) {
            return this.renderAction(this.styles.leftAction, this.props.left);
        } else {
            return this.renderButtons(this.styles.leftAction, this.props.left);
        }
    };

    renderRightActions = (): ReactNode => {
        if (this.isRightDisabled) {
            return undefined;
        } else if (this.isRightSideAction) {
            return this.renderAction(this.styles.rightAction, this.props.right);
        } else {
            return this.renderButtons(this.styles.rightAction, this.props.right);
        }
    };

    updateRef = (ref: any) => {
        this.row = ref;
    };

    close = () => {
        this.row.close();
    };

    private renderAction = (style: PanelStyle, content: ReactNode) => (
        <RectButton style={style} onPress={this.close}>
            {content}
        </RectButton>
    );

    private renderButtons = (style: PanelStyle, content: ReactNode) => {
        const actionStyle = { ...style };
        delete actionStyle.panelSize; // Deleting this property to avoid warnings
        return (
            <View style={{ width: style.panelSize, flexDirection: "row" }}>
                <View style={actionStyle}>{content}</View>
            </View>
        );
    };

    private onSwipeLeft(): void {
        if (this.isLeftSideAction && this.props.onSwipeLeft && this.props.onSwipeLeft.canExecute) {
            if (this.props.closeOnFinishLeft) {
                this.row.close();
            }
            this.props.onSwipeLeft.execute();
        }
    }

    private onSwipeRight(): void {
        if (this.isRightSideAction && this.props.onSwipeRight && this.props.onSwipeRight.canExecute) {
            if (this.props.closeOnFinishRight) {
                this.row.close();
            }
            this.props.onSwipeRight.execute();
        }
    }
}
