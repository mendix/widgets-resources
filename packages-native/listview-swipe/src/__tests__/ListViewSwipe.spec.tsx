import { createElement } from "react";
import { View, Text } from "react-native";
import { render } from "react-native-testing-library";
import { ListViewSwipe } from "../ListViewSwipe";
import { ListViewSwipeProps } from "../../typings/ListViewSwipeProps";
import { ListViewSwipeStyle } from "../ui/styles";
import { actionValue } from "../../../util-widgets/test";

jest.mock("NativeModules", () => ({
    UIManager: {
        RCTView: () => ({
            directEventTypes: {}
        })
    },
    KeyboardObserver: {},
    RNGestureHandlerModule: {
        attachGestureHandler: jest.fn(),
        createGestureHandler: jest.fn(),
        dropGestureHandler: jest.fn(),
        updateGestureHandler: jest.fn(),
        Direction: {
            RIGHT: 1,
            LEFT: 2,
            UP: 4,
            DOWN: 8
        },
        State: { BEGAN: "BEGAN", FAILED: "FAILED", ACTIVE: "ACTIVE", END: "END" }
    },
    PlatformConstants: {
        forceTouchAvailable: false
    }
}));

describe("List View Swipe", () => {
    let defaultProps: ListViewSwipeProps<ListViewSwipeStyle>;

    beforeEach(() => {
        defaultProps = {
            style: [],
            content: null,
            leftRenderMode: "action",
            left: (
                <View>
                    <Text>Test</Text>
                </View>
            ),
            closeOnFinishLeft: false,
            right: null,
            rightRenderMode: "disabled",
            closeOnFinishRight: false
        };
    });

    it("renders", () => {
        const component = render(<ListViewSwipe {...defaultProps} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it("render with left and action", () => {
        const onChangeAction = actionValue();
        const left = (
            <View>
                <Text>Test</Text>
            </View>
        );
        const component = render(<ListViewSwipe {...defaultProps} left={left} onSwipeLeft={onChangeAction} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it("render with right and action", () => {
        const onChangeAction = actionValue();
        const right = (
            <View>
                <Text>Test</Text>
            </View>
        );
        const component = render(<ListViewSwipe {...defaultProps} right={right} onSwipeRight={onChangeAction} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it("render with  left and right with actions", () => {
        const onChangeAction = actionValue();
        const left = (
            <View>
                <Text>Left Side</Text>
            </View>
        );
        const right = (
            <View>
                <Text>Right Side</Text>
            </View>
        );
        const component = render(
            <ListViewSwipe
                {...defaultProps}
                left={left}
                right={right}
                onSwipeLeft={onChangeAction}
                onSwipeRight={onChangeAction}
            />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
});
