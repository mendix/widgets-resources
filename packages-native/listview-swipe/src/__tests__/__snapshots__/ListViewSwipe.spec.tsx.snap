// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List View Swipe render with archive animation 1`] = `
<View
  style={Object {}}
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <RNGestureHandlerButton
        activeOpacity={0.105}
        collapsable={false}
        onActiveStateChange={[Function]}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        onPress={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              false,
            ],
          ]
        }
        testID="list-view-swipe-test$leftAction"
        underlayColor="black"
      >
        <View
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View>
          <Text>
            Test
          </Text>
        </View>
      </RNGestureHandlerButton>
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row-reverse",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      pointerEvents="auto"
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": -0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
          }
        }
        testID="list-view-swipe-test"
      />
    </View>
  </View>
</View>
`;

exports[`List View Swipe render with buttons option 1`] = `
<View
  style={Object {}}
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "width": 128,
          }
        }
        testID="list-view-swipe-test$leftButtons"
      >
        <View
          style={
            Object {
              "alignItems": "stretch",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <RNGestureHandlerButton
            activeOpacity={0.105}
            collapsable={false}
            onActiveStateChange={[Function]}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
            underlayColor="black"
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "bottom": 0,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
            Button 1
          </RNGestureHandlerButton>
          <RNGestureHandlerButton
            activeOpacity={0.105}
            collapsable={false}
            onActiveStateChange={[Function]}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
            underlayColor="black"
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "bottom": 0,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
            Button 2
          </RNGestureHandlerButton>
        </View>
      </View>
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row-reverse",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      pointerEvents="auto"
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": -0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
          }
        }
        testID="list-view-swipe-test"
      />
    </View>
  </View>
</View>
`;

exports[`List View Swipe render with left and action 1`] = `
<View
  style={Object {}}
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <RNGestureHandlerButton
        activeOpacity={0.105}
        collapsable={false}
        onActiveStateChange={[Function]}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        onPress={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              false,
            ],
          ]
        }
        testID="list-view-swipe-test$leftAction"
        underlayColor="black"
      >
        <View
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View>
          <Text>
            Test
          </Text>
        </View>
      </RNGestureHandlerButton>
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row-reverse",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      pointerEvents="auto"
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": -0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
          }
        }
        testID="list-view-swipe-test"
      />
    </View>
  </View>
</View>
`;

exports[`List View Swipe render with left and right with actions 1`] = `
<View
  style={Object {}}
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <RNGestureHandlerButton
        activeOpacity={0.105}
        collapsable={false}
        onActiveStateChange={[Function]}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        onPress={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              false,
            ],
          ]
        }
        testID="list-view-swipe-test$leftAction"
        underlayColor="black"
      >
        <View
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View>
          <Text>
            Left Side
          </Text>
        </View>
      </RNGestureHandlerButton>
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row-reverse",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      pointerEvents="auto"
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": -0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
          }
        }
        testID="list-view-swipe-test"
      />
    </View>
  </View>
</View>
`;

exports[`List View Swipe render with right and action 1`] = `
<View
  style={Object {}}
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <RNGestureHandlerButton
        activeOpacity={0.105}
        collapsable={false}
        onActiveStateChange={[Function]}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        onPress={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              false,
            ],
          ]
        }
        testID="list-view-swipe-test$leftAction"
        underlayColor="black"
      >
        <View
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View>
          <Text>
            Test
          </Text>
        </View>
      </RNGestureHandlerButton>
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row-reverse",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      pointerEvents="auto"
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": -0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
          }
        }
        testID="list-view-swipe-test"
      />
    </View>
  </View>
</View>
`;

exports[`List View Swipe render with toggle animation 1`] = `
<View
  style={Object {}}
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <RNGestureHandlerButton
        activeOpacity={0.105}
        collapsable={false}
        onActiveStateChange={[Function]}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        onPress={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "flex": 0,
                "width": 128,
              },
            ],
          ]
        }
        testID="list-view-swipe-test$leftAction"
        underlayColor="black"
      >
        <View
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View>
          <Text>
            Test
          </Text>
        </View>
      </RNGestureHandlerButton>
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row-reverse",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      pointerEvents="auto"
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": -0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
          }
        }
        testID="list-view-swipe-test"
      />
    </View>
  </View>
</View>
`;

exports[`List View Swipe renders 1`] = `
<View
  style={Object {}}
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <RNGestureHandlerButton
        activeOpacity={0.105}
        collapsable={false}
        onActiveStateChange={[Function]}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        onPress={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              false,
            ],
          ]
        }
        testID="list-view-swipe-test$leftAction"
        underlayColor="black"
      >
        <View
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View>
          <Text>
            Test
          </Text>
        </View>
      </RNGestureHandlerButton>
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row-reverse",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      pointerEvents="auto"
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": -0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
          }
        }
        testID="list-view-swipe-test"
      />
    </View>
  </View>
</View>
`;
